plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts

```diff
@@ -149,9 +153,25 @@ export default function transformProps(chartProps: ChartProps): EchartsProps {
     xAxis: {
       type: 'time',
       axisLabel: {
-        showMinLabel: xAxisShowMinLabel,
-        showMaxLabel: xAxisShowMaxLabel,
-        formatter: xAxisFormatter,
+        // force to display showMinLabel and showMaxLabel
+        // formatter can control their visibility
+        showMinLabel: true,
+        showMaxLabel: true,
+        formatter:
+          typeof xAxisFormatter === 'undefined'
+            ? xAxisFormatter
+            : function (value: number) {
+                const isMinLabel = timestamps[0] === value;
+                const isMaxLabel = timestamps[timestamps.length - 1] === value;
+                if (isMinLabel) {
+                  return xAxisShowMinLabel ? xAxisFormatter(value) : '';
+                }
+                if (isMaxLabel) {
+                  return xAxisShowMaxLabel ? xAxisFormatter(value) : '';
+                }
+
+                return xAxisFormatter(value);
+              },
```

Comment:
I don't think we need to explicitly remove the min/max values if the checkboxes are unchecked. I only think we need to make sure that the min and max values are showing if the checkboxes are checked. Therefore, I think the only change required here is
```js
axisLabel: {
    showMinLabel: !!xAxisShowMinLabel,
    showMaxLabel: !!xAxisShowMaxLabel,
}
```
to make sure that if the control value is 
- `undefined` or `false`: let ECharts decide what ticks to show.
`true`: forces showing the min/max values.

@zhaoyongjie wha do you think?

Reply:
`undefined ` and `null` are `auto` for ECharts.  And `false` does not force to be not displayed.

@junlincc Do you have any suggestions?

Reply:
sorry for being late to the party! please correct me if i understand the discussion inaccurately 
`undefined` = initial state, `null` = value does not exist, `false` = unchecking the box correct? 
what does `auto` do? 

Proposing expected behavior
1. have both checkboxes checked and showing min & max on the x-axis by default when users entering Explore the first time
2. changes should be sticky- 1) unchecking the boxes -> hide values, change should persist/sticky for users 2) rechecking boxes -> show values, change persist